### PR TITLE
Replace the global investigation with detectives

### LINKED ISSUE #4: Configuration should not use globals
When running an investigation, you don't want to configure the global `Political.octokit` and `Policial::StyleGuides::Ruby.config_file` just before, especially if you're on a multi-threaded application.

I think there should be a way to configure those for only one investigation. I think the config file is only necessary for `run`, and the octokit client for `accuse`. It could be that we pass to the method, or we could set up the investigation itself with those parameters.

### PR DESCRIPTION
**Pull request by volmer**
Introducing `Policial::Detective` :tada: 

Instead of running investigations with a global `Octokit` client, now you'll have to instantiate a `Policial::Detective` first with your credentials. `Policial::Detective` is pretty much a mix of `OctokitClient` and `Policial::Investigation`, both removed by this PR.

For more info on how Policial will work from now on, please check out the new _Usage_ section on [`README`](https://github.com/volmer/policial/blob/policial-client/README.md#usage).

Closes #4, #5

@etiennebarrie any thoughts on this?

### CODE REVIEW

File: lib/policial/client.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+module Policial
+  # Public: Starting with an Octokit client and a pull request,
+  # it checks all changes introduced looking for style guide violations. It
+  # also accuses all present violations through comments on all relevant lines
+  # in the pull request.
+  class Client
```
Comment (rafaelfranca) [inline lib/policial/client.rb]: This classes could be renamed to something else to better intention revealing. This `Client` class is not just a client. It knows how to investigate a pull request and know about violations and know what accuse. Maybe `Policial::Detective`?

**Comment by volmer** [comment]
Renamed `Client` to `Detective`. Thanks @rafaelfranca ! <3

File: lib/policial/client.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      violations.each do |violation|
+        if accusation_policy.allowed_for?(violation)
+          commenter.comment_violation(violation)
+        end
+      end
+    end
+
+    def pull_request=(pull_request)
+      @pull_request = pull_request
+      @pull_request.octokit = @octokit
```
Comment (rafaelfranca) [inline lib/policial/client.rb]: It feels weird to have this side effect of assigning the pull request. I was thinking if we could change this method to a factory method that will create the `PullRequest` object. Something like:

``` ruby
def file(content)
  @pull_request = PullRequest.new(content.merge(octokit: ocktokit)
```

And the usage flow would be simpler, and users would just have to know about the `Detective` class and `PullRequest` could be changed to be internal API.

``` ruby
policial_client = Policial::Client.new(octokit)
policial_client.file(repo: 'volmer/my_repo', number: 3, head_sha: 'headsha')
```

Comment (volmer) [inline lib/policial/client.rb]: Indeed, that's so much better. I've added `Detective#brief` in order to accomplish that (`#file` might cause confusion with `PullRequest#files`). Thanks!

**Comment by rafaelfranca** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by arthurnn** [comment]
so, what do we need to change on policial.io?

**Comment by volmer** [comment]
@arthurnn I just opened a PR for that https://github.com/arthurnn/policial.io/pull/10

**Pull request merged**